File: Python/Rebill/Create_Recurring_Payment_CC.py
##
# BluePay Python Sample code.
#
# This code sample creates a recurring payment charging $15.00 per month for one year.
##

import os.path, sys
sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), os.pardir))
from BluePay import BluePay

account_id = "Merchant's Account ID Here"
secret_key = "Merchant's Secret Key Here"
mode = "TEST"  

payment = BluePay(
    account_id = account_id, 
    secret_key = secret_key, 
    mode = mode
)

payment.set_customer_information(
    name1 = "Bob",
    name2 = "Tester",
    addr1 = "123 Test St.",
    addr2 = "Apt #500",
    city = "Testville",
    state = "IL",
    zipcode = "54321",
    country = "USA"
)

payment.set_cc_information(
    card_number = "4111111111111111",
    card_expire = "1215",
    cvv2 = "123"
)

payment.auth(amount = '0.00')

# Set Recurring Payment Information
payment.set_rebilling_information(
  reb_first_date = "2015-01-01", # Rebill Start Date: Jan. 1, 2015
  reb_expr = "1 MONTH", # Rebill Frequency: 1 MONTH
  reb_cycles = "12", # Rebill # of Cycles: 12
  reb_amount = "15.00" # Rebill Amount: $15.00
)

# Makes the API Request
payment.process()

# Read response from BluePay
if payment.is_successful_response():
    print 'Transaction ID: ' + payment.trans_id_response
    print 'Rebill ID: ' + payment.reb_id_response
    print 'Transaction Status: ' + payment.status_response
    print 'Transaction Message: ' + payment.message_response
    print 'Transaction AVS Result: ' + payment.avs_code_response
    print 'Transaction CVV2 Result: ' + payment.cvv2_code_response
    print 'Masked Payment Account: ' + payment.masked_account_response
    print 'Transaction Auth Code: ' + payment.auth_code_response
else:
    print payment.message_response

File: Perl/Rebill/Create_Recurring_Payment_CC.pl
##
# BluePay Perl Sample code.
#
# This code sample creates a recurring payment charging $15.00 per month for one year.
##

use strict;
use lib '..';
use bluepay;

my $account_id = "100228390579";
my $secret_key = "AKGIF9X9WT9CLQCWDFONC8N3HXRL9Y5K";
my $mode = "TEST";

my $rebill = BluePay->new(
	$account_id, 
	$secret_key, 
	$mode
);

$rebill->set_customer_information({
	first_name => 'Bob',
	last_name => 'Tester', 
	address1 => '123 Test St.',
	address2 => 'Apt', 
	city => 'Testville', 
	state => 'IL', 
	zip_code => '54321',
	country => 'USA',
	phone => '123-123-12345',
	email => 'test@bluepay.com'
});

$rebill->set_cc_information({
	cc_number =>'4111111111111111', # Customer Credit Card Number
	cc_expiration => '0815', # Card Expiration Date: MMYY
	cvv2 =>'123' # Card CVV2
});

$rebill->set_recurring_payment({
	reb_first_date => "2015-01-01", # Rebill Start Date: Jan. 1, 2015
	reb_expr => "1 MONTH", # Rebill Frequency: 1 MONTH
	reb_cycles => "12", # Rebill # of Cycles: 12
	reb_amount => "15.00" # Rebill Amount: $15.00
});

$rebill->auth({amount => '0.00'}); # Card authorization amount: $0.00

# Makes the API Request with BluePay
$rebill->process();

# Reads the response from BluePay
if ($rebill->is_successful_response()){
	print "TRANSACTION ID: " . $rebill->{RRNO} . "\n";
	print "REBILL ID: " . $rebill->{REBID} . "\n";
	print "TRANSACTION STATUS: " . $rebill->{Result} . "\n";
	print "TRANSACTION MESSAGE: " . $rebill->{MESSAGE} . "\n";
	print "AVS RESULT: " .$rebill->{AVS} . "\n";
	print "CVV2 RESULT: " . $rebill->{CVV2} . "\n";
	print "MASKED PAYMENT ACCOUNT: " . $rebill->{PAYMENT_ACCOUNT} . "\n";
	print "CARD TYPE: " . $rebill->{CARD_TYPE} . "\n";
	print "AUTH CODE: " . $rebill->{AUTH_CODE} . "\n";
} else {
	print $rebill->{MESSAGE} . "\n";
}